### ISSUE TITLE
Error requiring `DiffEqBase` from `Latexify`

### ISSUE THREAD

**Issue by cadojo** [korsbo/Latexify.jl#258]
Using `ModelingToolkit` and/or `Symbolics` results in an error, seemingly while importing `Latexify`. This happens when I `]add AstrodynamicalModels#main`, then enter `using AstrodynamicalModels`.

<details>

```
┌ Warning: Error requiring `DiffEqBase` from `Latexify`
│   exception =
│    LoadError: UndefVarError: ParameterizedFunctions not defined
│    Stacktrace:
│      [1] top-level scope
│        @ ~/.julia/packages/Latexify/L5vkp/ext/ParameterizedFunctionsExt.jl:4
│      [2] include(mod::Module, _path::String)
│        @ Base ./Base.jl:419
│      [3] include(x::String)
│        @ Latexify ~/.julia/packages/Latexify/L5vkp/src/Latexify.jl:1
│      [4] macro expansion
│        @ ~/.julia/packages/Requires/Z8rfN/src/Requires.jl:40 [inlined]
│      [5] top-level scope
│        @ ~/.julia/packages/Latexify/L5vkp/src/Latexify.jl:69
│      [6] eval
│        @ ./boot.jl:368 [inlined]
│      [7] eval
│        @ ~/.julia/packages/Latexify/L5vkp/src/Latexify.jl:1 [inlined]
│      [8] (::Latexify.var"#179#191")()
│        @ Latexify ~/.julia/packages/Requires/Z8rfN/src/require.jl:101
│      [9] macro expansion
│        @ timing.jl:382 [inlined]
│     [10] err(f::Any, listener::Module, modname::String, file::String, line::Any)
│        @ Requires ~/.julia/packages/Requires/Z8rfN/src/require.jl:47
│     [11] (::Latexify.var"#178#190")()
│        @ Latexify ~/.julia/packages/Requires/Z8rfN/src/require.jl:100
│     [12] withpath(f::Any, path::String)
│        @ Requires ~/.julia/packages/Requires/Z8rfN/src/require.jl:37
│     [13] (::Latexify.var"#177#189")()
│        @ Latexify ~/.julia/packages/Requires/Z8rfN/src/require.jl:99
│     [14] #invokelatest#2
│        @ ./essentials.jl:729 [inlined]
│     [15] invokelatest
│        @ ./essentials.jl:726 [inlined]
│     [16] foreach(f::typeof(Base.invokelatest), itr::Vector{Function})
│        @ Base ./abstractarray.jl:2774
│     [17] loadpkg(pkg::Base.PkgId)
│        @ Requires ~/.julia/packages/Requires/Z8rfN/src/require.jl:27
│     [18] #invokelatest#2
│        @ ./essentials.jl:729 [inlined]
│     [19] invokelatest
│        @ ./essentials.jl:726 [inlined]
│     [20] run_package_callbacks(modkey::Base.PkgId)
│        @ Base ./loading.jl:869
│     [21] _tryrequire_from_serialized(modkey::Base.PkgId, path::String, sourcepath::String, depmods::Vector{Any})
│        @ Base ./loading.jl:944
│     [22] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String, build_id::UInt64)
│        @ Base ./loading.jl:1028
│     [23] _require(pkg::Base.PkgId)
│        @ Base ./loading.jl:1315
│     [24] _require_prelocked(uuidkey::Base.PkgId)
│        @ Base ./loading.jl:1200
│     [25] macro expansion
│        @ ./loading.jl:1180 [inlined]
│     [26] macro expansion
│        @ ./lock.jl:223 [inlined]
│     [27] require(into::Module, mod::Symbol)
│        @ Base ./loading.jl:1144
│     [28] eval
│        @ ./boot.jl:368 [inlined]
│     [29] eval_user_input(ast::Any, backend::REPL.REPLBackend)
│        @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:151
│     [30] repl_backend_loop(backend::REPL.REPLBackend)
│        @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:247
│     [31] start_repl_backend(backend::REPL.REPLBackend, consumer::Any)
│        @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:232
│     [32] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool)
│        @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:369
│     [33] run_repl(repl::REPL.AbstractREPL, consumer::Any)
│        @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:355
│     [34] (::Base.var"#967#969"{Bool, Bool, Bool})(REPL::Module)
│        @ Base ./client.jl:419
│     [35] #invokelatest#2
│        @ ./essentials.jl:729 [inlined]
│     [36] invokelatest
│        @ ./essentials.jl:726 [inlined]
│     [37] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
│        @ Base ./client.jl:404
│     [38] exec_options(opts::Base.JLOptions)
│        @ Base ./client.jl:318
│     [39] _start()
│        @ Base ./client.jl:522
│    in expression starting at /Users/joey/.julia/packages/Latexify/L5vkp/ext/ParameterizedFunctionsExt.jl:1
└ @ Requires ~/.julia/packages/Requires/Z8rfN/src/require.jl:51
```

</details>

**Comment by ChrisRackauckas** [comment]
This should be fixed by https://github.com/korsbo/Latexify.jl/pull/259

**Comment by korsbo** [comment]
This should be sorted by https://github.com/korsbo/Latexify.jl/releases/tag/v0.15.20
Please let me know if the problem persists.

**Comment by hexaeder** [comment]
```
~ main ⇡5 ⇢5 !14 ❯ julia --startup-file=no                                                      26s
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.9.0-rc2 (2023-04-01)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.9) pkg> activate --temp
  Activating new project at `/var/folders/cx/04cwlxl9287_ndx30k6q3n1h0000gn/T/jl_4PEHo8`

(jl_4PEHo8) pkg> add ModelingToolkit
```
still errors on me with Latexify 15.20... sure the [diff to version 15.20](https://github.com/korsbo/Latexify.jl/compare/v0.15.19...v0.15.20) contains the right code?

**Comment by KristofferC** [comment]
I think https://github.com/korsbo/Latexify.jl/pull/260 is right.

**Comment by hexaeder** [comment]
Im confused. Isn't `ParameterizedFunctionsExt` extending `ParameterizedFunctions`? In That case 

https://github.com/korsbo/Latexify.jl/blob/e7b195fb5d679da7e15b4c405505039229367436/Project.toml#L26

should be changed. If #260 is correct it looks more like it is extending `DiffEqBase` in which case the Requires code

https://github.com/korsbo/Latexify.jl/blob/e7b195fb5d679da7e15b4c405505039229367436/src/Latexify.jl#L68-L70

should be probably adapted to match the Pkg extensions code (and trigger on `DiffEqBase`)

**Comment by KristofferC** [comment]
Original code had this code trigger from DiffEqBase:

https://github.com/korsbo/Latexify.jl/blob/ddf49bdc304ccf9f36f00ab10276f29eed70f8f8/src/Latexify.jl#L57-L59

**Comment by ChrisRackauckas** [comment]
> Im confused. Isn't ParameterizedFunctionsExt extending ParameterizedFunctions? In That case

It "kind of" is, kind of in that this was downstreamed to ParameterizedFunctions.jl years ago, which then changed to the ModelingToolkit form, and so it already has this overload defined (more correctly) here:

https://github.com/SciML/ParameterizedFunctions.jl/blob/master/src/latexify.jl

So the code here was throwing a warning, not used, and incorrect 😅 since April 2020 https://github.com/SciML/ParameterizedFunctions.jl/commit/6871a878637b3ac53b3226b306116ca9544e5b32. Right answer is to just delete it.

Similarly, DiffEqBiology was renamed Catalyst in April 2020 https://github.com/SciML/Catalyst.jl/releases/tag/v5.0.0 and with that change used ModelingToolkit as its underpinning, and it again has its own Latexify recipes which are more correct than the ones that were here https://github.com/SciML/Catalyst.jl/blob/master/src/latexify_recipes.jl .

So... the right answer is just to delete those since they existed at the start of Latexify to get adoption, but shortly afterwards the downstream started building recipes correctly and we just forgot to remove the original plugin code.

**Comment by korsbo** [comment]
I agree that the support should just be dropped. Indeed - I don't particularly like keeping any package-specific latexify code here. No one is active in maintaining them here and most of the current ones are pretty obsolete.

v0.16.0 sorted that out but was released as a breaking change (because it was) so it's not available to users before other packages have updated the compat bounds. In retrospect, it might have been worth to violate semver to make sure that the fix reached users immediately

I was hoping that the 0.15.20 backport would sort the issue out but I released it without much testing so maybe not. 

For the backport bugfix - I don't think it makes much sense to load the extension when someone pulls in `DiffEqBase` into their workspace. The extension is specific to `ParameterizedFunctions.jl` and should not be loaded unless someone loads that (obsolete) package.

**Comment by KristofferC** [comment]
> The extension is specific to ParameterizedFunctions.jl and should not be loaded unless someone loads that (obsolete) package.

But that's how it used to work (ref https://github.com/korsbo/Latexify.jl/issues/258#issuecomment-1521984270)? Can't we just go back to the working version first?

**Comment by korsbo** [comment]
I just released 0.15.21 where I followed @KristofferC's advice to just fix how the loading of DiffEqBase loads the ParameterizedFunctions extension. This seems to have resolved problems locally. This backports a fix such that we don't mess up installations across the ecosystem. The better fix for this though has already been released as 0.16.0 where we just drop support. Please let me know if problems persists with 0.15.21
